http/src/request/applications/mod.rs

```diff
@@ -0,0 +1,64 @@
+use std::{
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
+};
+
+mod create_guild_command;
+mod delete_guild_command;
+mod get_guild_commands;
+mod set_guild_commands;
+mod update_guild_command;
+
+mod create_global_command;
+mod delete_global_command;
+mod get_global_commands;
+mod set_global_commands;
+mod update_global_command;
+
+mod interaction_callback;
+
+pub use create_guild_command::CreateGuildCommand;
+pub use delete_guild_command::DeleteGuildCommand;
+pub use get_guild_commands::GetGuildCommands;
+pub use set_guild_commands::SetGuildCommands;
+pub use update_guild_command::UpdateGuildCommand;
+
+pub use create_global_command::CreateGlobalCommand;
+pub use delete_global_command::DeleteGlobalCommand;
+pub use get_global_commands::GetGlobalCommands;
+pub use set_global_commands::SetGlobalCommands;
+pub use update_global_command::UpdateGlobalCommand;
+
+pub use interaction_callback::InteractionCallback;
+
+/// The error created if the creation of interacion fails.
+#[derive(Debug)]
+pub struct InteractionError {
+    pub(crate) kind: InteractionErrorType,
+}
+
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum InteractionErrorType {
+    ApplicationIdNotPresent,
+}
+
+impl InteractionError {
```

Comment:
I think this needs the other two methods that are common elsewhere

Reply:
The reason I left them out is that they would not be able to be used for anything now and probably not in the future either.

Reply:
The only reason I mentioned it is there is precedent for other http methods without a source having these implemented. If someone were to write a generic error handler that calls these methods on http request builders that could create errors, that relied on those methods, they wouldn't be present on this impl.